### PR TITLE
[WIP] improve the librispeech recipe

### PR DESCRIPTION
**Pull request by sw005320**
Now I'm working on the improvement of the librispeech recipe motivated by the RWTH setup (thanks to Rohit Prabhavalkar and Kazuki Irie)
- sentencepiece model as a default
- VGG-BLSTM
- shallow and wide network (3layer BLSTMS with 1024 units (encoder), 1024 dim attention, unidirectional LSTM with 1024 units (decoder)
- fast convergence (change the maximum of epochs from 15 to 10)
- significant WER improvement (from 7.2 to 5.1 in `test_clean`)

TODO
- [ ] change the decoder layer from 1 to 2
- [ ] tune some search parameters (rnnlm weight)
- [ ] upload a model (#322)

### CODE REVIEW

**Comment by chenzhehuai** [comment]
It is consistent with my observation except that adding more layers up to 6 with 1024 units still obtains some improvement.

**Comment by sw005320** [comment]
Thanks. Will follow your suggestions (when GPUs are available). Could you share your WER results? If it is better than https://github.com/espnet/espnet/blob/d51e76c0baa556e28a3e090335944478828fbc65/egs/librispeech/asr1/RESULTS, then, I may follow your network architecture, or ask you to make a PR.

**Comment by sw005320** [comment]
The trained models can be provided through the release (e.g., https://github.com/espnet/espnet/releases/download/untagged-f5ccde023841a43380a9/librispeech_asr1.tgz)

**Comment by chenzhehuai** [comment]
No, my observation is from train_100. I think you system is the best til now.

**Comment by sw005320** [comment]
https://arxiv.org/pdf/1805.03294
This is not exactly same as what we're now using, but our set up is based on the discussion with them what would be most dominant given our current implementation.

**Comment by ruizhilijhu** [comment]
From this paper, their system played with the subsampling factor 32 in the pertaining and 8 for fine-tuning and they showed some improvement. 

In our babel-10 setting, we used subsampling factor 4, and we allowed the minimum number of frames in an utterance to be 10 frames. 

Is there any specific reason use our current setting?

**Comment by sw005320** [comment]
We don't have a specific reason, and we may test further subsampling like the RWTH paper, but I internally found that further subsampling (8) slightly degrades the performance in the Librispeech task.
